### PR TITLE
Reward logging

### PR DESCRIPTION
**Pull request by aaronschachter**
If the `dosomething_reward_log` variable is set, log lots of messages around automatic creation of Rewards and Shipments.

Refs https://jira.dosomething.org/browse/DS-219

### CODE REVIEW

File: lib/modules/dosomething/dosomething_reward/dosomething_reward.module [hunk 1]
```diff
@@ -228,18 +228,45 @@ function dosomething_reward_reportback_count($reportback) {
   $uid = $reportback->uid;
   $account = user_load($uid);
 
+  $log = variable_get('dosomething_reward_log');
+  $log_type = 'dosomething_reward_reportback_count';
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_reward/dosomething_reward.module]: Would it be more explicit to name it `$log_category`? I understand, [`watchdog()`](https://api.drupal.org/api/drupal/includes%21bootstrap.inc/function/watchdog/7) calls it `$type`, but it's also

> `$type`: The **category** to which this message belongs.

File: lib/modules/dosomething/dosomething_reward/dosomething_reward.module [hunk 2]
```diff
@@ -228,18 +228,45 @@ function dosomething_reward_reportback_count($reportback) {
   $uid = $reportback->uid;
   $account = user_load($uid);
 
+  $log = variable_get('dosomething_reward_log');
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_reward/dosomething_reward.module]: I'd make it more clear:

``` php
$log = variable_get('dosomething_reward_log', FALSE) ? TRUE : FALSE;
```

File: lib/modules/dosomething/dosomething_reward/includes/dosomething_reward.inc [hunk 3]
```diff
@@ -21,7 +21,13 @@ class RewardEntity extends Entity {
    * Sets the Reward as redeemed and creates a relevant Shipment if applciable.
    */
   public function redeem($values = array()) {
+    $log = variable_get('dosomething_reward_log');
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_reward/includes/dosomething_reward.inc]: Same here:

``` php
$log = variable_get('dosomething_reward_log', FALSE) ? TRUE : FALSE;
```

File: lib/modules/dosomething/dosomething_reward/dosomething_reward.module (hunk 2 above)
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_reward/dosomething_reward.module]: Consider defining it as a constant on the top of `dosomething_reward.module`.
I doubt it'll change its value during script runtime.

Thus you'll be able to reuse it in `dosomething_reward.inc`.

Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_reward/dosomething_reward.module]: Yeah that's a really good idea.

**Comment by sergiitk** [comment]
Great! :+1:

**Pull request merged**